agent/agent.go

```diff
@@ -349,6 +359,9 @@ func New(bd BaseDeps) (*Agent, error) {
 		router:          bd.Router,
 	}
 
+	// Initialize the UI Config
+	a.uiConfig.Store(a.config.UIConfig)
```

Comment:
Did you want to add something like `Clone()` on the UIConfig so you can do a deep copy of these things before stuffing them into a shared atomic? The enclosing struct is not a pointer so the top level will shallow copy, but these fields will not:
```
MetricsProviderFiles       []string
MetricsProxy               UIMetricsProxy    // nested: AddHeaders []UIMetricsProxyAddHeader
DashboardURLTemplates      map[string]string
```

Reply:
Is there a reason to believe we need a copy? Config should generally only be modified by the user and used without modification.

Reply:
Unclear, I just get a little twitchy sometimes with shared concurrent state being handed off like this without seeing all of the ways it might be used (since this PR does not include the use of this stuff, just the config of the stuff).

Reply:
Yeah I generally assume that nothing will be badly behaved enough to change config structs directly outside of the `config` package.